### PR TITLE
update icons to PNGs; use avatars and fix overflow in tlf list

### PR DESCRIPTION
**Pull request by songgao**
Also refactored the icon stuff a little bit.

This PR is made against `jzila/KBFS-2756-show-in-finder`; Will retarget to master after John's branch is merged.

### CODE REVIEW

File: shared/constants/types/fs.js [hunk 1]
```diff
@@ -211,3 +204,49 @@ export const sortSettingToIconTypeAndText = (s: _SortSetting): sortSettingDispla
...
+      iconColor: string,
+    }
+
+export type ItemStyles = {
+  iconSpec: PathItemIconSpec,
+  textColor: string,
+  textType: TextType,
+}
+
+export const makeBasicPathItemIconSpec = (iconType: IconType, iconColor: string): PathItemIconSpec => ({
```
Comment (jzila) [inline shared/constants/types/fs.js]: these functions should probably be in `constants/fs.js`

File: shared/fs/path-item-icon.js [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+import {Avatar, Icon} from '../common-adapters'
+
+type PathItemIconProps = {
+  spec: Types.PathItemIconSpec,
+}
+
+const PathItemIcon = ({spec}: PathItemIconProps) => {
+  switch (spec.type) {
+    case 'teamAvatar':
+      return <Avatar size={32} teamname={spec.teamName} isTeam={true} style={iconStyle} />
```
Comment (jzila) [inline shared/fs/path-item-icon.js]: These sizes will need to be different on mobile, won't they? Did you test this on mobile?

File: shared/fs/path-item-icon.js [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+    case 'avatar':
+      return <Avatar size={32} username={spec.username} style={iconStyle} />
+    case 'avatars':
+      // Use first avatar for now.
+      // TODO: fix this when we have support for three avatars as in design.
+      return <Avatar size={32} username={spec.usernames[0]} style={iconStyle} />
+    case 'basic':
+      return <Icon type={spec.iconType} style={{...iconStyle, fontSize: 32, iconColor: spec.iconColor}} />
```
Comment (jzila) [inline shared/fs/path-item-icon.js]: Might as well factor out this `{{...}}` style.

File: shared/fs/index.js [hunk 4]
```diff
@@ -156,4 +145,16 @@ const stylesRowRightBox = {
...
 
+const iconPlaceholderIconStyle = {
+  marginRight: globalMargins.small,
+  fontSize: 32,
+}
+
+const rowTextStyles = memoize(color => ({
+  color,
+  overflow: 'hidden',
+  textOverflow: 'ellipsis',
```
Comment (chrisnojima) [inline shared/fs/index.js]: this and whitespace might not work in rn, just a heads up

File: shared/fs/path-item-icon.js (hunk 2 above)
Comment (songgao) [reply shared/fs/path-item-icon.js]: Good point. I haven't tested this on mobile yet; was planning to do that altogether in my popup menu ticket.

File: shared/fs/index.js (hunk 4 above)
Comment (songgao) [reply shared/fs/index.js]: Oh hmm what would a good alternative be?

Comment (chrisnojima) [reply shared/fs/index.js]: i don't remember offhand, you can look around in the code but likely you want to use isMobile ? {} : {}

Comment (songgao) [reply shared/fs/index.js]: Got it thanks!

**Comment by songgao** [comment]
@chrisnojima would you mind thumb this for now, and I'll fix the mobile thing in a separate PR?

**Pull request merged**